earth_enterprise/src/fusion/portableglobe/cutter/cgi-bin/core/glc_assembler.py

```diff
@@ -196,11 +208,21 @@ def ExtractFileFromGlx(self, glc_path, file_in_glc, output_file):
 
   def ExtractFilelistFromGlx(self, glc_path, regex):
     """Extract given file from glc file and save it to the given path."""
+    logger = self.GetLogger()
     os_cmd = ("%s/geglxinfo "
               "--glx=\"%s\" "
               "--list_files"
               % (COMMAND_DIR, glc_path))
     glx_info = utils.RunCmd(os_cmd)
+    if len(glx_info) == 0:
```

Comment:
could 'glx_info' be null triggering an exception when we try to get its length ?

Reply:
glx_info is expected to be a list.  I'm not sure what len() does with a None object.  I *think* both an empty list and a None object evaluate to 'False' so maybe `if not glx_info:` is safer?  Or I could do a compound if statement if you feel that is more understandable.  I'll try a few things and see what works.  But if you feel a compound if statement is still the better thing to do I'll do just do: `if glx_info is None or len(glx_info) ==0:`.  Question: does Python have "short circuit if statement"?

Reply:
C++ does stop if the first test fails but not sure how python does. May be try it out or google it.

Reply:
According to this it does: http://bioruffo.net/lpotf/short-circuit-evaluations-in-python/188/

Reply:
I verified using the interpreter that yes it does support short circuit statements.  And a None object evaluates to false in an if statement and so does an empty list.   But unlike an empty string it looks like an empty list evaluates to true in an if statement.  So I can do a compound if statement to guard against None return.